src/crypto/CHIPCryptoPAL.cpp

```diff
@@ -97,25 +113,31 @@ CHIP_ERROR ReadDerUnsignedIntegerIntoRaw(Reader & reader, uint8_t * raw_integer_
     // We now have the rest of the tag that is a "minimal length" unsigned integer.
     // Blit it at the correct offset, since the order we use is MSB first for
     // both ASN.1 and EC curve raw points.
-    size_t offset = raw_integer_length - integer_len;
-    return reader.ReadBytes(&raw_integer_out[offset], integer_len).StatusCode();
+    size_t offset = raw_integer_out.size() - integer_len;
+    return reader.ReadBytes(raw_integer_out.data() + offset, integer_len).StatusCode();
 }
 
-size_t EmitDerIntegerFromRaw(const uint8_t * raw_integer, size_t raw_integer_length_bytes, uint8_t * out_der_integer,
-                             size_t out_der_integer_size)
+/**
+ * @brief Utility to emit a DER-encoded INTEGER given a raw unsigned large integer
+ *        in big-endian order.
+ * @param[in] raw_integer Bytes of a large unsigned integer in big-endian, possibly including leading zeroes
+ * @param[out] out_der_integer Buffer to receive the DER-encoded integer
+ * @return The non-zero length used by DER encoding in `out_der_ingeger`, or 0 on error, such as value too large to fit output.
+ */
+size_t EmitDerIntegerFromRaw(const ByteSpan & raw_integer, MutableByteSpan out_der_integer)
 {
-    if ((raw_integer == nullptr) || (raw_integer_length_bytes == 0))
+    if ((raw_integer.empty()) || (raw_integer.size() == 0))
```

Comment:
```suggestion
    if (raw_integer.empty())
```
The implementation of `empty()` is: `size() == 0`

Reply:
I mistook `FixedSpan::empty()` (which is `== nullptr`)